reccmp/isledecomp/compare/core.py

```diff
@@ -710,36 +711,24 @@ def _unique_names_for_overloaded_functions(self):
         """Our asm sanitize will use the "friendly" name of a function.
         Overloaded functions will all have the same name. This function detects those
         cases and gives each one a unique name in the db."""
-        repeat_names: dict[str, list[tuple[int, str | None]]] = {}
-
-        # Select addresses and symbols for all repeated function names
-        for recomp_addr, name, symbol in self._db.sql.execute(
-            """SELECT recomp_addr, json_extract(kvstore,'$.name') as name, json_extract(kvstore,'$.symbol')
-            from entities where name in (
-                select json_extract(kvstore,'$.name') as name from entities
-                where json_extract(kvstore,'$.type') = ?
-                group by name having count(name) > 1
-            )""",
-            (EntityType.FUNCTION,),
-        ):
-            # TODO: Thunk's link to the original function is lost once the record is created.
-            if "Thunk of" in name:
-                continue
-
-            repeat_names.setdefault(name, []).append((recomp_addr, symbol))
-
         with self._db.batch() as batch:
-            for name, items in repeat_names.items():
-                for i, (recomp_addr, symbol) in enumerate(items, start=1):
-                    # Just number it to start, in case we don't have a symbol.
-                    new_name = f"{name}({i})"
+            for func in get_overloaded_functions(self._db):
+                # TODO: Thunk's link to the original function is lost once the record is created.
+                if "Thunk of" in func.name:
+                    continue
+
+                # Just number it to start, in case we don't have a symbol.
+                new_name = f"{func.name}({func.sequence})"
 
-                    if symbol is not None:
-                        dm_args = get_function_arg_string(symbol)
-                        if dm_args is not None:
-                            new_name = f"{name}{dm_args}"
+                if func.symbol is not None:
+                    dm_args = get_function_arg_string(func.symbol)
+                    if dm_args is not None:
+                        new_name = f"{func.name}{dm_args}"
```

Comment:
Do we have/need any precautions in case the function arg string collides for two variants? Personally, I think the odds are very small, just something to think about.